sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -594,7 +594,12 @@ class SessionCatalog(
             child = parser.parsePlan(viewText))
           SubqueryAlias(table, child, Some(name.copy(table = table, database = Some(db))))
         } else {
-          SubqueryAlias(table, SimpleCatalogRelation(metadata), None)
+          val tableRelation = CatalogRelation(
+            metadata,
+            // we assume the data columns are always nullable.
+            metadata.dataSchema.asNullable.toAttributes,
+            metadata.partitionSchema.toAttributes)
```

Comment:
Do we need to call `asNullable` for partitionSchema? In the original `SimpleCatalogRelation`, we did it for `output`